app/frontend/wallet/cardano-wallet.js

```diff
@@ -26,15 +25,14 @@ function txFeeFunction(txSizeInBytes) {
 }
 
 const CardanoWallet = async (options) => {
-  const {mnemonicOrHdNodeString, config, randomSeed, network} = options
-
+  const {walletSecretDef, config, randomSeed, network} = options
   const state = {
     randomSeed: randomSeed || Math.floor(Math.random() * MAX_INT32),
     ownUtxos: {},
     overallTxCountSinceLastUtxoFetch: 0,
     accountIndex: HARDENED_THRESHOLD,
     network,
-    derivationScheme: options.derivationScheme || derivationSchemes.v2,
+    derivationScheme: null, //is set in cryptoproviders
```

Comment:
I would not set any derivationscheme there. I would maintain the pattern that derivation scheme should be managed solely by crypto providers. So let's say the address manager below should not receive the derivation scheme as a parameter, but rather ask the cryptoprovider about the derivation scheme. This will avoid mistakenly setting one derivation scheme in one module and another scheme in other module.